webui/react/src/e2e/models/hew/DataGrid.ts

```diff
@@ -92,6 +99,95 @@ export class DataGrid<
     return this.columnHeight;
   }
 
+  /**
+   * Scolls the table
+   * @param {object} args - The x and y coordinates
+   * @param {number} [args.xAbsolute] - The x coordinate
+   * @param {number} [args.xRelative] - The relative x coordinate
+   */
+  private async scrollTable(args: { xAbsolute: number } | { xRelative: number }): Promise<void> {
+    const box = await this.pwLocator.boundingBox();
+    if (box === null) {
+      throw new Error('Expected to see a bounding box for the table.');
+    }
+    const page = this.root._page;
+    // move mouse to the center of the table
+    await page.mouse.move((box.x + box.width) / 2, (box.y + box.height) / 2);
+    // scroll the table
+    await page.mouse.wheel(
+      'xAbsolute' in args ? args.xAbsolute : Math.max(box.width - args.xRelative, 200),
+      0,
+    );
+    // scrolling isn't waited on, and in this instance it's not guaranteed either. let's just wait a bit.
+    await page.waitForTimeout(3_000);
```

Comment:
well typically we'd want to wait on the columns to change, but the columns wont change if the last column is already in view

Reply:
yeah the problem is this is too strict `checkColumnInView`

the best we could do is "wait for columns to change" but even that wouldn't be guaranteed. All we can do is wait for a change to "maybe" happen, and then check with ```
      if (JSON.stringify(indexes) === JSON.stringify(prevIndexes)) {
        return false;
      }
```

Reply:
im just gonna leave this here since i considered it, but at this moment i can't figure out what else I'd wait on anyway.
```
  /**
   * Scolls the table
   * @param {object} obj
   * @param {number} [obj.xAbsolute] - The x coordinate. If not provided, the table will incrementally scroll to the right.
   * @param {Function} [obj.waitMethod] - The method we use to wait for the table to scroll. Default is to wait 3 seconds.
   */
  private async scrollTable({ xAbsolute, waitMethod = async () => await this.root._page.waitForTimeout(3000) }: { xAbsolute?: number, waitMethod?: () => Promise<void> } = {}): Promise<void> {
    const box = await this.pwLocator.boundingBox();
    if (box === null) {
      throw new Error('Expected to see a bounding box for the table.');
    }
    const page = this.root._page;
    // move mouse to the center of the table
    await page.mouse.move((box.x + box.width) / 2, (box.y + box.height) / 2);
    // scroll the table
    await page.mouse.wheel(
      xAbsolute ? xAbsolute : box.width - 450,
      0,
    );
    await waitMethod();
  }
```